### PR TITLE
Refs #33517 -- Prevented second lookup from returning fractional seconds on PostgreSQL.

### PR DESCRIPTION
**Pull request by nabil-rady**
Not sure if the tests are written properly (the need for a new class and the usage of psycopg2.extras). But the test correctly detects the error before the fix.

### CODE REVIEW

File: tests/postgres_tests/test_operations.py [hunk 1]
```diff
@@ -581,3 +589,20 @@ def test_validate(self):
...
+    def test_date_extraction(self):
+        r = Room.objects.create(number=1)
+        hr = HotelReservation.objects.create(
+            room=r,
+            datespan=DateRange(lower="2022-2-21", upper="2022-2-22"),
+            start=datetime.datetime(2022, 2, 21, 2, 36, 10, 10000),
+            end=datetime.datetime(2022, 2, 21, 2, 36, 10, 20000),
+        )
+        qs = HotelReservation.objects.filter(start__second=F("end__second"))
+        self.assertSequenceEqual(qs, [hr])
```
Comment (felixxm) [inline tests/postgres_tests/test_operations.py:608]: This issue is PostgreSQL-specific, but should work on all builtin backends. I'd remove these tests and add a database agnostic test to the `tests.db_functions.datetime.test_extract_trunc`, e.g.
```diff
diff --git a/tests/db_functions/datetime/test_extract_trunc.py b/tests/db_functions/datetime/test_extract_trunc.py
index 67a4b81766..682f78f149 100644
--- a/tests/db_functions/datetime/test_extract_trunc.py
+++ b/tests/db_functions/datetime/test_extract_trunc.py
@@ -899,6 +899,20 @@ class DateFunctionTests(TestCase):
             2,
         )
 
+    def test_extract_second_func_no_fractional(self):
+        start_datetime = datetime(2015, 6, 15, 14, 30, 50, 321)
+        end_datetime = datetime(2016, 6, 15, 14, 30, 50, 783)
+        if settings.USE_TZ:
+            start_datetime = timezone.make_aware(start_datetime)
+            end_datetime = timezone.make_aware(end_datetime)
+        obj = self.create_model(start_datetime, end_datetime)
+        self.assertSequenceEqual(
+            DTModel.objects.filter(
+                start_datetime__second=F("end_datetime__second"),
+            ),
+            [obj],
+        )
+
     def test_trunc_func(self):
         start_datetime = datetime(999, 6, 15, 14, 30, 50, 321)
         end_datetime = datetime(2016, 6, 15, 14, 10, 50, 123)
```

**Review by felixxm — COMMENTED** [review]
@nabil-rady Thanks :+1:

**Comment by nabil-rady** [comment]
@felixxm Done

**Review by felixxm — COMMENTED** [review]
@nabil-rady Thanks for updates :+1: Unfortunately we have the same issue on Oracle, but I have no idea how to fix it so I skipped these tests for now.

**Pull request merged**